[Bug] legal code translation handling is inconsistent/unreliable

## Description
legal code translation handling is not reliable
1. Sometimes coverage tests fail:
    ```
    ======================================================================
    FAIL: test_legal_code_translation_by_40_es (legal_tools.tests.test_views.ViewLegalCodeTest)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/usr/local/lib/python3.9/unittest/case.py", line 59, in testPartExecutor
        yield
      File "/usr/local/lib/python3.9/unittest/case.py", line 592, in run
        self._callTestMethod(testMethod)
      File "/usr/local/lib/python3.9/unittest/case.py", line 550, in _callTestMethod
        method()
      File "/home/cc/cc-legal-tools-app/legal_tools/tests/test_views.py", line 792, in test_legal_code_translation_by_40_es
        self.assertContains(rsp, "Sección 1 – Definiciones")
      File "/usr/local/lib/python3.9/site-packages/django/test/testcases.py", line 471, in assertContains
        self.assertTrue(real_count != 0, msg_prefix + "Couldn't find %s in response" % text_repr)
      File "/usr/local/lib/python3.9/unittest/case.py", line 688, in assertTrue
        raise self.failureException(msg)
    AssertionError: False is not true : Couldn't find 'Sección 1 – Definiciones' in response
    
    ----------------------------------------------------------------------
    ```
   - See: [`c3f1d03093` `legal_tools/tests/test_views.py` lines 803-830](https://github.com/creativecommons/cc-legal-tools-app/blob/c3f1d03093b88b48bdf803109806ec2d45e8d842/legal_tools/tests/test_views.py#L803-L830)
2. Deeds & UX strings are inconsistent for published files
   - For example, `creativecommons/cc-legal-tools-data: docs/licenses/by/4.0/legalcode.mi.html`: [Lines 6-8](https://github.com/creativecommons/cc-legal-tools-data/blob/0e98c495385a496579d15e058236657c7b8881a0/docs/licenses/by/4.0/legalcode.mi.html#L7):
    ```html
      <title>
       CC BY 4.0 Legal Code — Attiecinājums 4.0 Starptautisks — Creative Commons
      </title>
    ```
     - The Deeds & UX translation for `mi`  Maori  (Māori) does not exceed threshold (`TRANSLATION_THRESHOLD`). This title should be in `en` English (`LANGUAGE_CODE`) instead of `lv` Latvian (latviešu)
     - This indicates a failure to adequate set/reset the language in `view_legal_code()`--`lv` Latvian (latviešu) is the language immediately before `mi`  Maori  (Māori)
     

## Reproduction
behavior is inconsistent 😢 

## Expectation
Legal Code languages should be
1. specified language
2. fallback to jurisdiction default language
3. fallback to default language (`LANGUAGE_CODE`): `en` English

I expect I need to update either or both of `get_translation_object()` and `active_translation()` in `i18n/utils.py`.

## Additional context
- legal code translation handling is bespoke due to the fact that Django does not currently support custom translation domains
  - [#6376 (Allow using custom gettext domains) – Django](https://code.djangoproject.com/ticket/6376)
    - Unfortunately work on that issue has stalled: [Refs #6376 -- Added support for custom gettext domains. by r4g3baby · Pull Request #10687 · django/django](https://github.com/django/django/pull/10687)
- Translation Code
  - [`class DjangoTranslation(gettext_module.GNUTranslations)`](https://github.com/django/django/blob/fdf209eab8949ddc345aa0212b349c79fc6fdebb/django/utils/translation/trans_real.py#L118) (Django 3.2.12)
    - [`class GNUTranslations(NullTranslations)`](https://github.com/python/cpython/blob/2de452f8bf2f78417e04bcf7919beb502c53a0e2/Lib/gettext.py#L358) (cpython 3.9.11)
      - [`class NullTranslations`](https://github.com/python/cpython/blob/2de452f8bf2f78417e04bcf7919beb502c53a0e2/Lib/gettext.py#L254) (cpython 3.9.11)
- Translation Documentation
  - [Translation | Django documentation | Django 3.2](https://docs.djangoproject.com/en/3.2/topics/i18n/translation/)
    - [The GNUTranslations class - gettext — Multilingual internationalization services — Python 3.9.11 documentation](https://docs.python.org/3.9/library/gettext.html#the-gnutranslations-class)
      - [The NullTranslations class - gettext — Multilingual internationalization services — Python 3.9.11 documentation](https://docs.python.org/3.9/library/gettext.html#the-nulltranslations-class)

--- a/i18n/tests/test_utils.py
+++ b/i18n/tests/test_utils.py
@@ -187,8 +187,8 @@ class TranslationTest(TestCase):
                 "i18n.utils.translation.trans_real.translation"
             ) as mock_trans:
                 result = get_translation_object(
-                    django_language_code="LANGUAGE_CODE",
                     domain="GETTEXT_DOMAIN",
+                    language_code="LANGUAGE_CODE",
                     language_default="LANGUAGE_DEFAULT",
                 )
         mock_djt.assert_called_with(
@@ -214,8 +214,8 @@ class TranslationTest(TestCase):
                 "i18n.utils.translation.trans_real.translation"
             ) as mock_trans:
                 result = get_translation_object(
-                    django_language_code="LANGUAGE_CODE",
                     domain="GETTEXT_DOMAIN",
+                    language_code="LANGUAGE_CODE",
                     language_default="LANGUAGE_DEFAULT",
                 )
         mock_djt.assert_called_with(
@@ -241,8 +241,8 @@ class TranslationTest(TestCase):
                 "i18n.utils.translation.trans_real.translation"
             ) as mock_trans:
                 result = get_translation_object(
-                    django_language_code="LANGUAGE_CODE",
                     domain="GETTEXT_DOMAIN",
+                    language_code="LANGUAGE_CODE",
                     language_default="LANGUAGE_DEFAULT",
                 )
         mock_djt.assert_called_with(

--- a/i18n/tests/test_utils.py
+++ b/i18n/tests/test_utils.py
@@ -172,7 +172,11 @@ class I18NTest(TestCase):
 
 
 class TranslationTest(TestCase):
-    def test_get_translation_object(self):
+    @override_settings(
+        LANGUAGES_MOSTLY_TRANSLATED=["LANGUAGE_CODE"],
+        LEGAL_CODE_LOCALE_PATH="LOCALE_DIRS",
+    )
+    def test_get_translation_object_specified_translated(self):
         translation_object = MagicMock()
 
         with mock.patch(
@@ -185,12 +189,68 @@ class TranslationTest(TestCase):
                 result = get_translation_object(
                     django_language_code="LANGUAGE_CODE",
                     domain="GETTEXT_DOMAIN",
+                    language_default="LANGUAGE_DEFAULT",
                 )
         mock_djt.assert_called_with(
             domain="GETTEXT_DOMAIN",
             language="LANGUAGE_CODE",
+            localedirs="LOCALE_DIRS",
         )
         mock_trans.assert_called_with("LANGUAGE_CODE")
+        self.assertEqual(translation_object, result)
+
+    @override_settings(
+        LANGUAGES_MOSTLY_TRANSLATED=["LANGUAGE_DEFAULT"],
+        LEGAL_CODE_LOCALE_PATH="LOCALE_DIRS",
+    )
+    def test_get_translation_object_default_translated(self):
+        translation_object = MagicMock()
+
+        with mock.patch(
+            "i18n.utils.translation.trans_real.DjangoTranslation"
+        ) as mock_djt:
+            mock_djt.return_value = translation_object
+            with mock.patch(
+                "i18n.utils.translation.trans_real.translation"
+            ) as mock_trans:
+                result = get_translation_object(
+                    django_language_code="LANGUAGE_CODE",
+                    domain="GETTEXT_DOMAIN",
+                    language_default="LANGUAGE_DEFAULT",
+                )
+        mock_djt.assert_called_with(
+            domain="GETTEXT_DOMAIN",
+            language="LANGUAGE_CODE",
+            localedirs="LOCALE_DIRS",
+        )
+        mock_trans.assert_called_with("LANGUAGE_DEFAULT")
+        self.assertEqual(translation_object, result)
+
+    @override_settings(
+        LANGUAGES_MOSTLY_TRANSLATED=[],
+        LEGAL_CODE_LOCALE_PATH="LOCALE_DIRS",
+    )
+    def test_get_translation_object_untranslated(self):
+        translation_object = MagicMock()
+
+        with mock.patch(
+            "i18n.utils.translation.trans_real.DjangoTranslation"
+        ) as mock_djt:
+            mock_djt.return_value = translation_object
+            with mock.patch(
+                "i18n.utils.translation.trans_real.translation"
+            ) as mock_trans:
+                result = get_translation_object(
+                    django_language_code="LANGUAGE_CODE",
+                    domain="GETTEXT_DOMAIN",
+                    language_default="LANGUAGE_DEFAULT",
+                )
+        mock_djt.assert_called_with(
+            domain="GETTEXT_DOMAIN",
+            language="LANGUAGE_CODE",
+            localedirs="LOCALE_DIRS",
+        )
+        mock_trans.assert_called_with(settings.LANGUAGE_CODE)
         self.assertEqual(translation_object, result)
 
     def test_active_translation(self):

--- a/i18n/utils.py
+++ b/i18n/utils.py
@@ -69,7 +69,7 @@ JURISDICTION_CURRENCY_LOOKUP = {
 # read and use the one that's there right now. Anyway, this site doesn't
 # need to perform all that well, since it just generates static files.
 def get_translation_object(
-    *, django_language_code: str, domain: str
+    *, django_language_code: str, domain: str, language_default: str
 ) -> translation.trans_real.DjangoTranslation:
     """
     Return a DjangoTranslation object suitable to activate when we're wanting
@@ -84,12 +84,23 @@ def get_translation_object(
     tool_translation_object = translation.trans_real.DjangoTranslation(
         language=django_language_code,
         domain=domain,
+        localedirs=settings.LEGAL_CODE_LOCALE_PATH,
     )
+
     # Add a fallback to the standard Django translation for this language. This
     # gets us the non-legal-code parts of the pages.
-    tool_translation_object.add_fallback(
-        translation.trans_real.translation(django_language_code)
-    )
+    if django_language_code in settings.LANGUAGES_MOSTLY_TRANSLATED:
+        tool_translation_object.add_fallback(
+            translation.trans_real.translation(django_language_code)
+        )
+    elif language_default in settings.LANGUAGES_MOSTLY_TRANSLATED:
+        tool_translation_object.add_fallback(
+            translation.trans_real.translation(language_default)
+        )
+    else:
+        tool_translation_object.add_fallback(
+            translation.trans_real.translation(settings.LANGUAGE_CODE)
+        )
 
     return tool_translation_object
 
@@ -123,9 +134,9 @@ def active_translation(
     yield
 
     if previous_translation is None:
-        del _active.value
+        translation.deactivate()
     else:
-        _active.value = previous_translation
+        translation.activate(previous_translation.language())
 
 
 def save_pofile_as_pofile_and_mofile(pofile: polib.POFile, pofile_path: str):

--- a/i18n/utils.py
+++ b/i18n/utils.py
@@ -64,12 +64,8 @@ JURISDICTION_CURRENCY_LOOKUP = {
 }
 
 
-# This function looks like a good candidate for caching, but we might be
-# changing the translated files while running and need to be sure we always
-# read and use the one that's there right now. Anyway, this site doesn't
-# need to perform all that well, since it just generates static files.
 def get_translation_object(
-    *, django_language_code: str, domain: str, language_default: str
+    domain: str, language_code: str, language_default: str
 ) -> translation.trans_real.DjangoTranslation:
     """
     Return a DjangoTranslation object suitable to activate when we're wanting
@@ -78,20 +74,24 @@ def get_translation_object(
 
     This fuction requires the legal code locales path to have been added to
     Django settings.LOCALE_PATHS
+
+    WARNING: this *does* make assumptions about the internals of Django's
+    translation system that could change on us.  It doesn't seem likely,
+    though.
     """
 
     # Start with a translation object for the domain for this tool.
     tool_translation_object = translation.trans_real.DjangoTranslation(
-        language=django_language_code,
+        language=language_code,
         domain=domain,
         localedirs=settings.LEGAL_CODE_LOCALE_PATH,
     )
 
     # Add a fallback to the standard Django translation for this language. This
     # gets us the non-legal-code parts of the pages.
-    if django_language_code in settings.LANGUAGES_MOSTLY_TRANSLATED:
+    if language_code in settings.LANGUAGES_MOSTLY_TRANSLATED:
         tool_translation_object.add_fallback(
-            translation.trans_real.translation(django_language_code)
+            translation.trans_real.translation(language_code)
         )
     elif language_default in settings.LANGUAGES_MOSTLY_TRANSLATED:
         tool_translation_object.add_fallback(
@@ -107,12 +107,13 @@ def get_translation_object(
 
 @contextmanager
 def active_translation(
-    translation_obj: translation.trans_real.DjangoTranslation,
+    translation_object: translation.trans_real.DjangoTranslation,
 ):
     """
     Context manager to do stuff within its context with a particular
     translation object set as the active translation.  (Use
-    ``get_translation_object`` to get a translation object to use with this.)
+    ``legal_code.get_translation_object()`` to get a translation object to use
+    with this.)
 
     Bypasses all the language code stuff that Django does when you use its
     ``activate(language_code)`` function.
@@ -124,19 +125,17 @@ def active_translation(
     translation system that could change on us.  It doesn't seem likely,
     though.
     """
-    _active = translation.trans_real._active
 
-    # Either _active.value points at a DjangoTranslation
-    # object, or _active has no 'value' attribute.
-    previous_translation = getattr(_active, "value", None)
-    _active.value = translation_obj
+    previous_language = translation.trans_real.get_language()
+    translation.trans_real._active.value = translation_object
 
     yield
 
-    if previous_translation is None:
-        translation.deactivate()
-    else:
-        translation.activate(previous_translation.language())
+    # The following logic is based on django.utils.translations.override()
+    if previous_language is not None:
+        translation.activate(previous_language)
+    else:  # pragma: no cover
+        translation.deactivate_all()
 
 
 def save_pofile_as_pofile_and_mofile(pofile: polib.POFile, pofile_path: str):

--- a/legal_tools/models.py
+++ b/legal_tools/models.py
@@ -357,9 +357,13 @@ class LegalCode(models.Model):
         return self.tool.resource_slug
 
     def get_translation_object(self):
+        language_default = get_default_language_for_jurisdiction(
+            self.tool.jurisdiction_code
+        )
         return get_translation_object(
             django_language_code=self.language_code,
             domain=self.tool.resource_slug,
+            language_default=language_default,
         )
 
     def get_pofile(self) -> polib.POFile:

--- a/legal_tools/models.py
+++ b/legal_tools/models.py
@@ -357,10 +357,9 @@ class LegalCode(models.Model):
         return self.tool.resource_slug
 
     def get_translation_object(self):
-        domain = self.tool.resource_slug
         return get_translation_object(
             django_language_code=self.language_code,
-            domain=domain,
+            domain=self.tool.resource_slug,
         )
 
     def get_pofile(self) -> polib.POFile:

--- a/legal_tools/models.py
+++ b/legal_tools/models.py
@@ -361,8 +361,8 @@ class LegalCode(models.Model):
             self.tool.jurisdiction_code
         )
         return get_translation_object(
-            django_language_code=self.language_code,
             domain=self.tool.resource_slug,
+            language_code=self.language_code,
             language_default=language_default,
         )
 

--- a/legal_tools/tests/test_models.py
+++ b/legal_tools/tests/test_models.py
@@ -258,7 +258,7 @@ class LegalCodeModelTest(TestCase):
         ) as mock_djt:
             legal_code.get_translation_object()
         mock_djt.assert_called_with(
-            django_language_code="de", domain="by-sa_40"
+            django_language_code="de", domain="by-sa_40", language_default="en"
         )
 
     def test_branch_name(self):

--- a/legal_tools/tests/test_models.py
+++ b/legal_tools/tests/test_models.py
@@ -258,7 +258,7 @@ class LegalCodeModelTest(TestCase):
         ) as mock_djt:
             legal_code.get_translation_object()
         mock_djt.assert_called_with(
-            django_language_code="de", domain="by-sa_40", language_default="en"
+            domain="by-sa_40", language_code="de", language_default="en"
         )
 
     def test_branch_name(self):

--- a/legal_tools/tests/test_views.py
+++ b/legal_tools/tests/test_views.py
@@ -709,6 +709,7 @@ class ViewLegalCodeTest(TestCase):
             tool__category="licenses",
             tool__base_url="https://creativecommons.org"
             "/licenses/by/3.0/de/",
+            tool__unit="by",
             tool__version="3.0",
             tool__jurisdiction_code="de",
         )
@@ -728,6 +729,7 @@ class ViewLegalCodeTest(TestCase):
         tool = ToolFactory(
             category="licenses",
             base_url="https://creativecommons.org/licenses/by/4.0/",
+            unit="by",
             version="4.0",
         )
         for language_code in ["ar", "es", settings.LANGUAGE_CODE]:
@@ -775,23 +777,21 @@ class ViewLegalCodeTest(TestCase):
         LANGUAGES_MOSTLY_TRANSLATED=["es", settings.LANGUAGE_CODE],
     )
     def test_view_legal_code_30_default_translated(self):
-        tool = ToolFactory(
-            category="licenses",
-            base_url="https://creativecommons.org/licenses/by/3.0/es/",
-            version="3.0",
-        )
         language_code = "ast"
         lc = LegalCodeFactory(
-            tool=tool,
+            tool__category="licenses",
+            tool__base_url="https://creativecommons.org/licenses/by/3.0/es/",
+            tool__unit="by",
+            tool__version="3.0",
+            tool__jurisdiction_code="es",
             language_code=language_code,
+            html="<html></html>",
         )
         url = lc.legal_code_url
         rsp = self.client.get(url)
         self.assertEqual(200, rsp.status_code)
         self.assertTemplateUsed(rsp, "legalcode.html")
-        self.assertTemplateUsed(
-            rsp, "includes/legalcode_licenses_3.0_unported.html"
-        )
+        self.assertTemplateUsed(rsp, "includes/legalcode_crude_html.html")
         context = rsp.context
         self.assertEqual(lc, context["legal_code"])
         self.assertContains(rsp, f'lang="{language_code}"')
@@ -801,23 +801,21 @@ class ViewLegalCodeTest(TestCase):
         LANGUAGES_MOSTLY_TRANSLATED=[settings.LANGUAGE_CODE],
     )
     def test_view_legal_code_30_default_untranslated(self):
-        tool = ToolFactory(
-            category="licenses",
-            base_url="https://creativecommons.org/licenses/by/3.0/es/",
-            version="3.0",
-        )
         language_code = "ast"
         lc = LegalCodeFactory(
-            tool=tool,
+            tool__category="licenses",
+            tool__base_url="https://creativecommons.org/licenses/by/3.0/es/",
+            tool__unit="by",
+            tool__version="3.0",
+            tool__jurisdiction_code="es",
             language_code=language_code,
+            html="<html></html>",
         )
         url = lc.legal_code_url
         rsp = self.client.get(url)
         self.assertEqual(200, rsp.status_code)
         self.assertTemplateUsed(rsp, "legalcode.html")
-        self.assertTemplateUsed(
-            rsp, "includes/legalcode_licenses_3.0_unported.html"
-        )
+        self.assertTemplateUsed(rsp, "includes/legalcode_crude_html.html")
         context = rsp.context
         self.assertEqual(lc, context["legal_code"])
         self.assertContains(rsp, f'lang="{language_code}"')
@@ -857,14 +855,13 @@ class ViewLegalCodeTest(TestCase):
     #     self.assertGreater(len(rsp.content.decode()), 0)
 
     def test_legal_code_translation_by_40_es(self):
-        tool = ToolFactory(
-            category="licenses",
-            base_url="https://creativecommons.org/licenses/by/4.0/",
-            version="4.0",
-        )
+        language_code = "es"
         legal_code = LegalCodeFactory(
-            tool=tool,
-            language_code="es",
+            tool__category="licenses",
+            tool__base_url="https://creativecommons.org/licenses/by/4.0/",
+            tool__unit="by",
+            tool__version="4.0",
+            language_code=language_code,
         )
         url = legal_code.legal_code_url
         rsp = self.client.get(url)

--- a/legal_tools/tests/test_views.py
+++ b/legal_tools/tests/test_views.py
@@ -741,7 +741,7 @@ class ViewLegalCodeTest(TestCase):
     @override_settings(
         LANGUAGES_MOSTLY_TRANSLATED=["ar", settings.LANGUAGE_CODE],
     )
-    def test_view_legal_code(self):
+    def test_view_legal_code_40(self):
         tool = ToolFactory(
             category="licenses",
             base_url="https://creativecommons.org/licenses/by/4.0/",
@@ -756,9 +756,30 @@ class ViewLegalCodeTest(TestCase):
             rsp = self.client.get(url)
             self.assertEqual(200, rsp.status_code)
             self.assertTemplateUsed(rsp, "legalcode.html")
-            self.assertTemplateUsed(
-                rsp, "includes/legalcode_licenses_4.0.html"
-            )
+            for template in (
+                "base.html",
+                "legalcode.html",
+                "includes/about_cc.html",
+                "includes/about_cc_and_license.html",
+                "includes/footer.html",
+                "includes/header.html",
+                "includes/legalcode_licenses_4.0.html",
+                "includes/legalcode_menu_sidebar.html",
+                "includes/licenses_header.html",
+                "includes/related_links.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/snippet/icon.html",
+                "includes/use_of_licenses.html",
+            ):
+                self.assertTemplateUsed(rsp, template)
             context = rsp.context
             self.assertEqual(lc, context["legal_code"])
             self.assertContains(rsp, f'lang="{language_code}"')
@@ -766,6 +787,58 @@ class ViewLegalCodeTest(TestCase):
                 self.assertContains(rsp, 'dir="ltr"')
             elif language_code == "ar":
                 self.assertContains(rsp, 'dir="rtl"')
+
+    @override_settings(
+        LANGUAGES_MOSTLY_TRANSLATED=["es", settings.LANGUAGE_CODE],
+    )
+    def test_view_legal_code_30_default_translated(self):
+        tool = ToolFactory(
+            category="licenses",
+            base_url="https://creativecommons.org/licenses/by/3.0/es/",
+            version="3.0",
+        )
+        language_code = "ast"
+        lc = LegalCodeFactory(
+            tool=tool,
+            language_code=language_code,
+        )
+        url = lc.legal_code_url
+        rsp = self.client.get(url)
+        self.assertEqual(200, rsp.status_code)
+        self.assertTemplateUsed(rsp, "legalcode.html")
+        self.assertTemplateUsed(
+            rsp, "includes/legalcode_licenses_3.0_unported.html"
+        )
+        context = rsp.context
+        self.assertEqual(lc, context["legal_code"])
+        self.assertContains(rsp, f'lang="{language_code}"')
+        self.assertContains(rsp, 'dir="ltr"')
+
+    @override_settings(
+        LANGUAGES_MOSTLY_TRANSLATED=[settings.LANGUAGE_CODE],
+    )
+    def test_view_legal_code_30_default_untranslated(self):
+        tool = ToolFactory(
+            category="licenses",
+            base_url="https://creativecommons.org/licenses/by/3.0/es/",
+            version="3.0",
+        )
+        language_code = "ast"
+        lc = LegalCodeFactory(
+            tool=tool,
+            language_code=language_code,
+        )
+        url = lc.legal_code_url
+        rsp = self.client.get(url)
+        self.assertEqual(200, rsp.status_code)
+        self.assertTemplateUsed(rsp, "legalcode.html")
+        self.assertTemplateUsed(
+            rsp, "includes/legalcode_licenses_3.0_unported.html"
+        )
+        context = rsp.context
+        self.assertEqual(lc, context["legal_code"])
+        self.assertContains(rsp, f'lang="{language_code}"')
+        self.assertContains(rsp, 'dir="ltr"')
 
     # NOTE: plaintext functionality disabled
     # def test_view_legal_code_plain_text(self):

--- a/legal_tools/tests/test_views.py
+++ b/legal_tools/tests/test_views.py
@@ -90,23 +90,6 @@ def expected_and_unexpected_strings_for_tool(tool):
     return expected, unexpected
 
 
-# All the valid units. They all start with "by", and have various
-# combinations of "nc", "nd", and "sa", in that order. But not all combinations
-# are valid, e.g. "nd" and "sa" are not compatible.
-units = []
-for bits in range(8):  # We'll enumerate the variations
-    parts = ["by"]
-    if bits & 1:
-        parts.append("nc")
-    if bits & 2:
-        parts.append("nd")
-    if bits & 4:
-        parts.append("sa")
-    if "nd" in parts and "sa" in parts:
-        continue  # Not compatible
-    units.append("-".join(parts))
-
-
 class ToolsTestsMixin:
     # Create some tools to test in setUp
     def setUp(self):

--- a/legal_tools/views.py
+++ b/legal_tools/views.py
@@ -497,11 +497,15 @@ def view_legal_code(
     request.path, language_code = normalize_path_and_lang(
         request.path, jurisdiction, language_code
     )
+    language_default = get_default_language_for_jurisdiction(jurisdiction)
     if language_code in settings.LANGUAGES_MOSTLY_TRANSLATED:
         translation.activate(language_code)
+    elif language_default in settings.LANGUAGES_MOSTLY_TRANSLATED:
+        translation.activate(language_default)
+    else:
+        translation.activate(settings.LANGUAGE_CODE)
 
     path_start = os.path.dirname(request.path)
-    language_default = get_default_language_for_jurisdiction(jurisdiction)
 
     # NOTE: plaintext functionality disabled
     # if is_plain_text:
@@ -528,7 +532,6 @@ def view_legal_code(
         tool,
     )
 
-    language_code = legal_code.language_code  # CC language code
     languages_and_links = get_languages_and_links_for_legal_codes(
         path_start=path_start,
         legal_codes=tool.legal_codes.all(),

--- a/legal_tools/views.py
+++ b/legal_tools/views.py
@@ -498,12 +498,6 @@ def view_legal_code(
         request.path, jurisdiction, language_code
     )
     language_default = get_default_language_for_jurisdiction(jurisdiction)
-    if language_code in settings.LANGUAGES_MOSTLY_TRANSLATED:
-        translation.activate(language_code)
-    elif language_default in settings.LANGUAGES_MOSTLY_TRANSLATED:
-        translation.activate(language_default)
-    else:
-        translation.activate(settings.LANGUAGE_CODE)
 
     path_start = os.path.dirname(request.path)
 
@@ -524,44 +518,53 @@ def view_legal_code(
         legal_code_url=request.path,
     )
 
+    # Use Deeds & UX translations for title instead of Legal Code
+    if language_code in settings.LANGUAGES_MOSTLY_TRANSLATED:
+        translation.activate(language_code)
+    elif language_default in settings.LANGUAGES_MOSTLY_TRANSLATED:
+        translation.activate(language_default)
+    else:
+        translation.activate(settings.LANGUAGE_CODE)
     tool = legal_code.tool
     tool_title = get_tool_title(tool)
 
-    category, category_title = get_category_and_category_title(
-        category,
-        tool,
-    )
-
-    languages_and_links = get_languages_and_links_for_legal_codes(
-        path_start=path_start,
-        legal_codes=tool.legal_codes.all(),
-        selected_language_code=language_code,
-    )
-
-    deed_rel_path = get_deed_rel_path(
-        legal_code.deed_url,
-        path_start,
-        language_code,
-        language_default,
-    )
-
-    kwargs = dict(
-        template_name="legalcode.html",
-        context={
-            "canonical_url": f"{settings.CANONICAL_SITE}{request.path}",
-            "category": category,
-            "category_title": category_title,
-            "deed_rel_path": deed_rel_path,
-            "identifier": tool.identifier(),
-            "languages_and_links": languages_and_links,
-            "legal_code": legal_code,
-            "tool": tool,
-            "tool_title": tool_title,
-        },
-    )
-
+    # Activate Legal Code translation
     current_translation = legal_code.get_translation_object()
     with active_translation(current_translation):
+
+        category, category_title = get_category_and_category_title(
+            category,
+            tool,
+        )
+
+        languages_and_links = get_languages_and_links_for_legal_codes(
+            path_start=path_start,
+            legal_codes=tool.legal_codes.all(),
+            selected_language_code=language_code,
+        )
+
+        deed_rel_path = get_deed_rel_path(
+            legal_code.deed_url,
+            path_start,
+            language_code,
+            language_default,
+        )
+
+        kwargs = dict(
+            template_name="legalcode.html",
+            context={
+                "canonical_url": f"{settings.CANONICAL_SITE}{request.path}",
+                "category": category,
+                "category_title": category_title,
+                "deed_rel_path": deed_rel_path,
+                "identifier": tool.identifier(),
+                "languages_and_links": languages_and_links,
+                "legal_code": legal_code,
+                "tool": tool,
+                "tool_title": tool_title,
+            },
+        )
+
         # NOTE: plaintext functionality disabled
         # if is_plain_text:
         #     response = HttpResponse(

